src/main.rs

```diff
@@ -45,32 +45,35 @@ fn main() {
 }
 
 fn check_path(arg: &str) -> Option<String> {
-    let path = Path::new(&arg);
-
-    match path.try_exists() {
-        Ok(true) => {
-            if path.is_file() {
-                match path.to_str() {
-                    Some(path_str) => Some(path_str.to_string()),
-                    None => process::exit(1),
-                }
-            } else {
-                match path.join("Cargo.lock").to_str() {
-                    Some(file_path) => check_path(file_path),
-                    None => {
-                        eprintln!("Error: failure appending Cargo.lock to {arg}");
-                        process::exit(1);
+    if arg == "-" {
+        Some(arg.to_string())
+    } else {
```

Comment:
With `if (..) { return ...; }`, the `else` block could be unwrapped, saving a level of indentation.

Which way do you prefer?